nemo/core/optim/lr_scheduler.py

```diff
@@ -832,8 +832,13 @@ def prepare_lr_scheduler(
         monitor = 'loss'
 
     # Store exact max_steps if it is provided
-    if 'max_steps' in scheduler_config and scheduler_config['max_steps'] is not None:
-        max_steps = scheduler_config['max_steps']
+    max_steps_from_cfg = scheduler_config.get('max_steps')
+    if max_steps_from_cfg is not None:
+        if max_steps_from_cfg == -1:
```

Comment:
what should we do if we want epoch based training? by default `max_steps` is set to -1, which means it'll return here before calculating the actual max_steps later in this function. `scheduler_config.get('max_steps')` will never be None, since otherwise PTL will throw an error saying max_teps must be an integer. @odelalleau @titu1994 @ericharper